### PR TITLE
Rewire notify by email

### LINKED ISSUE #32494: Remove button to notify users for local shares
Since recent OC versions (10.0.8 or 10.0.9), an email is already sent to recipients through the OC notifications framework. This obsoletes the need for this button.

Would be good if we could remove it at least in OC 11 as it would save a lot of refactoring work that is needed to remove the old Sharing 1.0 code. See https://github.com/owncloud/core/pull/26608

@pmaier1

### PR DESCRIPTION
**Pull request by VicDeo**
## Description
- [x] Get rid of legacy `API::register` invocations in favor of routing
- [x]  Kill an ugly OCSShareWrapper
- [x] Refactor `Share20OCS` to be a controller
- [x] Move 2 endpoints from **core/ajax/share.php** into the new controller
- [x] Rewrite the implementation for these endpoints in order not to use legacy sharing methods

## Related Issue
- Fixes https://github.com/owncloud/core/issues/32494
- unblocks https://github.com/owncloud/core/pull/26608

## Motivation and Context
Sharing code cleanup

## How Has This Been Tested?
1. share with user
2. click `notify by email`
3. check that email has been arrived
4. reload the page and check that there is no  `notify by email` button any more
5. Login as user and check that the emailed permalink works

1. share with group
2. click `notify by email`
3. check that email for all users with emails have been arrived
4. reload the page and check that there is no  `notify by email` button any more
5. Login as any user from this group and check that permalink  works (TBH I didn't test them all) 

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Technical debt
- [ ] Tests

## Checklist:
- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added

## Open tasks:
- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)
Since it doesn't beak anything a backport would be nice

### CODE REVIEW

File: apps/files_sharing/lib/AppInfo/Application.php [hunk 1]
```diff
@@ -75,6 +77,24 @@ public function __construct(array $urlParams = []) {
 			);
 		});
 
+		$container->registerService('Share20OcsController', function (SimpleContainer $c) use ($server) {
+			return new Share20OcsController(
```
[the remark this answers is not in the document]
Comment (VicDeo) [reply apps/files_sharing/lib/AppInfo/Application.php]: @PVince81 it can't resolve `\OCP\IUser` to inject current user properly

**Comment by VicDeo** [comment]
Ready for review

**Review by PVince81 — COMMENTED** [review]
Awesome!

Please check why codecov is complaining (good luck getting an actual report page if it's in the right mood).

Also please state how you confirmed that it still worked as required by the issue template. Would be good to confirm that the button as clicked in the web UI still behaves as before.

**Comment by PVince81** [comment]
ok, so codecov issue related to the code that declares the new routes, not testable.
routes should already be covered by acceptance API tests

**Review by PVince81 — APPROVED** [review]
:+1:

**Pull request merged**